var/spack/repos/builtin/packages/magma/package.py

```diff
@@ -46,6 +46,7 @@ class Magma(CMakePackage, CudaPackage, ROCmPackage):
     depends_on("cuda@8:", when="@2.5.1: +cuda")  # See PR #14471
     depends_on("hipblas", when="+rocm")
     depends_on("hipsparse", when="+rocm")
+    depends_on("python", when="@2.7.1:", type="build")
```

Comment:
Only repo checkouts require Python. Regular releases do not as all the files are already generated and included in the tar-ball. It should  be `when=@master`. Just see my `plasma/package.py` for a similar dependency on Python: `when=@develop`. Also, no version of version of Python is specified: I doubt the code generator works with Python 3 and its syntax changes (in PLASMA, we had to test with both 2 and 3).

Reply:
> It should be when=@master

Thanks, pushed

> Also, no version of version of Python is specified: I doubt the code generator works with Python 3 and its syntax changes


Hm - what is the correct python version dependency? My build went through fine with spack's default - python-3.10. And it doesn't support python2 anymore 

 Perhaps magma needs updates to work properly with python3? [and also change from looking for 'python' to 'python3'?]

Reply:
I see a comment that the python files in magma use as source whatever is in PLASMA and to modify there first, if needed, e.g., [codegen.py](https://github.com/icl-utk-edu/plasma/blob/main/tools/codegen.py). The comment says "Tested with python 2.7.9 and 3.4.3." and I just tested with Python 2.7.13. I guess if there is need to change the default python in MAGMA we could do it, e.g., if it is modified first in PLASMA so that the python scripts for code generation remain the same?